### PR TITLE
Sagonzal/app creation script

### PR DESCRIPTION
**Pull request by sangonzal**
- Update to call Microsoft Graph
- Add app creation scripts
- Update README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,76 +1,124 @@
...
-This sample demonstrates a command line application calling a web API that is secured using Azure AD. The Java application uses the Active Directory Authentication Library for Java (ADAL4J) to obtain a JWT access token through the OAuth 2.0 protocol. The access token is sent to the web API to authenticate the user. This sample shows you how to use ADAL to authenticate users via raw credentials (username and password) via a text-only interface.
+### Overview
 
+This sample demonstrates a Java console application calling Microsoft Graph that is secured using Azure Active Directory.
+
+1. The Java application uses the Active Directory Authentication Library for Java (ADAL4J) to obtain a JWT access token through the OAuth 2.0 protocol.
+2. The access token is used as a bearer token to authenticate the user when calling the Microsoft Graph.
+
+### Scenario: 
+
```
Comment (jmprieur) [inline README.md]: you might want to add a picture with the topology (See @abhidnya13 who has done it for one of her's)

File: README.md [hunk 2]
```diff
@@ -1,76 +1,124 @@
...
+### Step 4:  Configure the sample to use your Azure AD tenant
 
-```
+In the steps below, ClientID is the same as Application ID or AppId.
 
-### Step 6: Package and then run the `public-client-adal4j-sample-jar-with-dependencies.jar ` file.
+#### Configure the app project
+
+1. Open the `src\main\java\PublicClient.java` file
+1. Find the app key `private final static String CLIENT_ID` and replace the existing value with the application ID (clientId) of the `Native-Headless-Application` application copied from the Azure portal.
```
Comment (jmprieur) [inline README.md]: Find the line ?

File: README.md [hunk 3]
```diff
@@ -1,76 +1,124 @@
...
+
+If you'd like to contribute to this sample, see CONTRIBUTING.md
 
-### Acknowledgements
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information, see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
 
-We would like to acknowledge the folks who own/contribute to the following projects for their support of Azure Active Directory and their libraries that were used to build this sample. In places where we forked these libraries to add additional functionality, we ensured that the chain of forking remains intact so you can navigate back to the original package. Working with such great partners in the open source community clearly illustrates what open collaboration can accomplish. Thank you!
+## More information
 
+For more information, see ADAL4J [conceptual documentation](https://github.com/AzureAD/azure-activedirectory-library-for-java/wiki)
```
Comment (jmprieur) [inline README.md]: Do you have a link on the specific flow with username/password in the wiki? 
if yo don't you might want to create it? (get inspired by the .NET one: [Acquiring tokens with username and password](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/wiki/Acquiring-tokens-with-username-and-password)

File: README.md [hunk 4]
```diff
@@ -1,76 +1,124 @@
...
-* `$ java -jar public-client-adal4j-sample-jar-with-dependencies.jar`
-
+- `$ java -jar public-client-adal4j-sample-jar-with-dependencies.jar`
 
 ### You're done!
 
-Your command line interface should prompt you for the username and password and then access the Graph API to retreive your user information.
+Your command line interface should prompt you for the username and password and then access the Microsoft Graph API to retrieve your user information.
+
+## Community Help and Support
```
Comment (jmprieur) [inline README.md]: I suggest you have a section "About the code", where you explain that the relevant code is line 45-47 ? (and maybe add it in a ```Java block

**Review by jmprieur — APPROVED** [review]
This is pretty good @sangonzal 
I've suggested a few improvements which would make this sample great.
don't hesitate to ping me (or check with @abhidnya13)

File: README.md (hunk 2 above)
Comment (sangonzal) [reply README.md]: Fixed!

File: README.md (hunk 3 above)
Comment (sangonzal) [reply README.md]: Information on username/password is in: https://github.com/AzureAD/azure-activedirectory-library-for-java/wiki/Acquiring-Tokens-with-username-and-password

File: README.md (hunk 4 above)
Comment (sangonzal) [reply README.md]: Done

File: AppCreationScripts/AppCreationScripts.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+### Presentation of the scripts
+
+This sample comes with two PowerShell scripts, which automate the creation of the Azure Active Directory applications, and the configuration of the code for this sample. Once you run them, you will only need to build the solution and you are good to test.
+
+These scripts are:
+
+- `Configure.ps1` which:
+  - creates Azure AD applications and their related objects (permissions, dependencies, secrets),
+  - changes the configuration files in the C# and JavaScript projects.
```
Comment (navyasric) [inline AppCreationScripts/AppCreationScripts.md]: This should be Java project

File: AppCreationScripts/AppCreationScripts.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+4. If you have never done it already, in the PowerShell window, install the AzureAD PowerShell modules. For this, type:
+    ```PowerShell
+    Install-Module AzureAD
+    ```
+5. Go to the `AppCreationScripts` sub-folder. From the folder where you cloned the repo,
+    ```PowerShell
+    cd AppCreationScripts
+    ```
+6. Run the scripts. See below for the [four options](#four-ways-to-run-the-script) to do that.
+7. Open the Visual Studio solution, and in the solution's context menu, choose **Set Startup Projects**.
```
Comment (navyasric) [inline AppCreationScripts/AppCreationScripts.md]: The Visual studio steps are not relevant for this sample

File: README.md [hunk 7]
```diff
@@ -1,76 +1,124 @@
...
-```java
-public class PublicClient {
+1. In the  **Azure Active Directory** pane, click on **App registrations** and choose **New application registration**.
+1. Enter a friendly name for the application, for example 'Native-Headless-Application' and select 'Native' as the *Application Type*.
+1. For the *Redirect URI*, enter `https://<your_tenant_name>/Native-Headless-Application`, replacing `<your_tenant_name>` with the name of your Azure AD tenant.
+1. Click on **Create** to create the application.
+1. In the succeeding page, Find the *Application ID* value and copy it to the clipboard. You'll need it to configure the Visual Studio configuration file for this project.
+1. Then click on **Settings**, and choose **Properties**.
+1. Configure Permissions for your application. To that extent, in the Settings menu, choose the 'Required permissions' section and then,
+   click on **Add**, then **Select an API**, and type `Microsoft Graph` in the textbox. Then, click on  **Select Permissions** and select **User.Read**.
```
Comment (navyasric) [inline README.md]: The manual step to 'Grant permissions' needs to be mentioned.

**Review by navyasric — CHANGES_REQUESTED** [review]
Sample works. In app generation scripts, cleanup script failed with this error: 
Cleanup : Cannot convert 'System.Object[]' to the type 'System.String' required by parameter 'ObjectId'. Specified method is not supported.

Suggested a few Readme updates.

File: README.md (hunk 1 above)
Comment (sangonzal) [reply README.md]: done

File: AppCreationScripts/AppCreationScripts.md (hunk 5 above)
Comment (sangonzal) [reply AppCreationScripts/AppCreationScripts.md]: Fixed

File: AppCreationScripts/AppCreationScripts.md (hunk 6 above)
Comment (sangonzal) [reply AppCreationScripts/AppCreationScripts.md]: Fixed

File: README.md (hunk 7 above)
Comment (sangonzal) [reply README.md]: done!

**Comment by jmprieur** [comment]
@sangonzal @navyasric : does the script work? do you need help?

**Review by navyasric — APPROVED** [review]

**Comment by navyasric** [comment]
@sangonzal Looks good to merge!

**Pull request merged**